_publications/jors/paper.tex

```diff
@@ -98,91 +96,171 @@ \section*{Introduction}
+avoids the complexity and administrative overhead of maintaining a
+live database and associated backend app. The PFHub infrastructure
+provides a template for other small scientific communities to host
+custom content and integrate data from members of their community.
+
+\begin{figure}
+  \includegraphics[width=4in]{pfhub_website.png}
+  \caption{Schematic overview of the PFHub framework for building
+    scientific research portals, simply.}
+  \centering
+  \label{fig:pfhub_website}
+\end{figure}
 
 \section*{Quality control}
 
-\textcolor{blue}{Detail the level of testing that has been carried out on the code (e.g. unit, functional, load etc.), and in which environments. If not already included in the software documentation, provide details of how a user could quickly understand if the software is working (e.g. providing examples of running the software with sample input and output data). }
+Every pull request to the repository is tested using Travis CI. The
+full test recipe is outlined in the \texttt{.travis.yml} file
+available in the repository~\cite{travisyml}. The entire framework is
+built using the Nix Package Manager~\cite{nix} and pinned to a
+specific version of the Nixpkgs repository. Using Nix ensures the
+development and automated testing environments are identical. PFHub is
+built using both Python and Coffeescript code so test and lint
+frameworks dedicated to both of those languages are used. Furthermore,
+Surge~\cite{surge} is used to display the final website link in the
+pull request so that reviewers can look at a visual display of the
+website before merging.
 
 \section*{(2) Availability}
 \vspace{0.5cm}
 \section*{Operating system}
 
-\textcolor{blue}{Please include minimum version compatibility.}
+PFHub can be deployed on any platform supporting Nix, which includes
+all contemporary Linux and Mac OS X platforms.
 
 \section*{Programming language}
 
-\textcolor{blue}{Please include minimum version compatibility.}
+PFHub is currently build and tested using the programming language and
+versions outlined in Table~\ref{tab:versions}.
+
+\begin{table}[h!]
+  \centering
+  \begin{tabular}{|l|l|}
+    \hline
+    Language         & Version \\
+    \hline
+    HTML             & 5       \\
+    Jupyter Notebook & 5.4.0   \\
+    JavaScript       & 5       \\
+    Nix              & 2.1.3   \\
+    CoffeeScript     & 1.12.7  \\
+    CSS              & 4       \\
+    \hline
+  \end{tabular}
+  \caption{PFHub programming languages and corresponding supported
+    versions.}
+  \label{tab:versions}
+\end{table}
+
 
 \section*{Additional system requirements}
 
-\textcolor{blue}{E.g. memory, disk space, processor, input devices, output devices.}
+There are no additionial system requirements.
 
 \section*{Dependencies}
 
-\textcolor{blue}{E.g. libraries, frameworks, incl. minimum version compatibility.}
+The entire environment can be built using Nix so the only required
+dependency is a functional Nix installation.
 
```

Comment:
This is only true because Nix pulls in the dependencies. It is also possible to build PFHub *outside* of Nix -- in which case, what dependencies must be satisfied? `yamllint, htmlproofer, jekyll, ...`